lib/ansible/modules/cloud/google/gce_template.py

```diff
@@ -0,0 +1,596 @@
+    description:
+      - The external IP address to use.
+        If 'ephemeral' (default), a new non-static address will be
+        used.  If 'None', then no external address will
+        be used.  To use an existing static IP address
+        specify adress name.
+    required: false
+    default: "ephemeral"
+  service_account_email:
+    description:
+      - service account email
+    required: false
+    default: null
+  service_account_permissions:
+    description:
+      - service account permissions (see
+        U(https://cloud.google.com/sdk/gcloud/reference/compute/instances/create),
+        --scopes section for detailed information)
+    required: false
+    default: null
+    choices: [
+      "bigquery", "cloud-platform", "compute-ro", "compute-rw",
+      "useraccounts-ro", "useraccounts-rw", "datastore", "logging-write",
+      "monitoring", "sql", "sql-admin", "storage-full", "storage-ro",
+      "storage-rw", "taskqueue", "userinfo-email"
+    ]
+  automatic_restart:
+    description:
+      - Defines whether the instance should be
+        automatically restarted when it is
+        terminated by Compute Engine.
+    required: false
+    default: null
+  preemptible:
+    description:
+      - Defines whether the instance is preemptible.
+    required: false
+    default: null
+  tags:
+    description:
+      - a comma-separated list of tags to associate with the instance
+    required: false
+    default: null
+  metadata:
+    description:
+      - a hash/dictionary of custom data for the instance;
+        '{"key":"value", ...}'
+    required: false
+    default: null
+  description:
+    description:
+      - description of instance template
+    required: false
+    default: null
+  disks:
+    description:
+      - a list of persistent disks to attach to the instance; a string value
+        gives the name of the disk; alternatively, a dictionary value can
+        define 'name' and 'mode' ('READ_ONLY' or 'READ_WRITE'). The first entry
+        will be the boot disk (which must be READ_WRITE).
+    required: false
+    default: null
+  nic_gce_struct:
+    description:
+      - Support passing in the GCE-specific
+        formatted networkInterfaces[] structure.
+    required: false
+    default: null
+  project_id:
+    description:
+      - your GCE project ID
+    required: false
+    default: null
+  pem_file:
+    description:
+      - path to the pem file associated with the service account email
+        This option is deprecated. Use 'credentials_file'.
+    required: false
+    default: null
+  credentials_file:
+    description:
+      - path to the JSON file associated with the service account email
+    default: null
+    required: false
+requirements:
+    - "python >= 2.6"
+    - "apache-libcloud >= 0.13.3, >= 0.17.0 if using JSON credentials,
+      >= 0.20.0 if using preemptible option"
+notes:
+  - JSON credentials strongly preferred.
+author: "Gwenael Pellen <gwenael.pellen@arkeup.com>"
```

Comment:
Please include your github ID in the author line, in between your name and email address.  For example:

`"my name (@my-github) <my-email@example.com>"`